tools/check-napi.js

```diff
@@ -0,0 +1,96 @@
+// Descend into a directory structure and, for each file matching *.node, output
+// based on the imports found in the file whether it's an N-API module or not.
+
+const fs = require('fs');
+const path = require('path');
+const child_process = require('child_process');
+
+// Read the output of the command, break it into lines, and use the reducer to
+// decide whether the file is an N-API module or not.
+function checkFile(file, command, arguments, reducer) {
+  const child = child_process.spawn(command, arguments, {
+    stdio: ['inherit', 'pipe', 'inherit']
+  });
+  let leftover = '';
+  let isNapi = undefined;
+  child.stdout.on('data', (chunk) => {
+    if (isNapi === undefined) {
+      chunk = leftover + chunk.toString();
+      const haveLeftover = !!chunk.match(/[\r\n]+$/);
+      chunk = chunk.split(/[\r\n]+/);
+      leftover = chunk.pop();
+      isNapi = chunk.reduce(reducer, isNapi);
+    }
+  });
+  child.on('exit', (code, signal) => {
+    if ((code === null && signal !== null) || (code !== 0)) {
+      console.log(
+        command + ' exited with code: '  + code + ' and signal: ' + signal);
+    } else {
+      console.log(
+          '\033[' + (isNapi ? '42' : '41') + 'm' +
+          (isNapi ? '    N-API' : 'Not N-API') +
+          '\033[0m: ' + file);
+    }
+  });
+}
+
+// Use nm -a to list symbols.
+function checkFileUNIX(file) {
+  checkFile(file, 'nm', ['-a', file], (soFar, line) => {
+      if (soFar === undefined) {
+        line = line.match(/([0-9a-f]*)? ([a-zA-Z]) (.*$)/);
+        line.shift();
+        if (line[1] === 'U') {
+          if (line[2].match(/^napi/)) {
```

Comment:
Wouldn't returning from here and changing the condition at 41 to `!soFar` be better?

Reply:
Also, if we don't want to get the matches but need to check if the string matches the format, then I believe `RegExp#test` would be better.

Reply:
I don't like having `return true` up here and then `return soFar` at the bottom. I'd rather have one return statement for the whole function.

The reason I don't have `!soFar` at the top is so if, in the future, we add a check that will disqualify a file from being N-API, we can set soFar to `false` and have the rest short-circuit.

Reply:
Fair enough 👍